contracts/cosmwasm-vm/cw-icon-light-client/src/light_client.rs

```diff
@@ -117,40 +126,55 @@ impl ILightClient for IconClient<'_> {
 
     fn update_client(
         &mut self,
+        caller: Addr,
         client_id: &str,
         signed_header: SignedHeader,
     ) -> Result<ConsensusStateUpdate, Self::Error> {
+        self.context.ensure_ibc_host(caller)?;
         let btp_header = signed_header.header.clone().unwrap();
+
         let mut state = self.context.get_client_state(client_id)?;
-        // let config = self.context.get_config()?;
 
-        if (btp_header.main_height - state.latest_height) > state.trusting_period {
+        if signed_header.trusted_height > btp_header.main_height {
+            return Err(ContractError::UpdateBlockOlderThanTrustedHeight);
+        }
+
+        let trusted_consensus_state = self
+            .context
+            .get_consensus_state(client_id, signed_header.trusted_height)?;
+
+        let current_proof_context_hash =
+            btp_header.get_next_proof_context_hash(&signed_header.current_validators);
+
+        if current_proof_context_hash != trusted_consensus_state.next_proof_context_hash {
+            return Err(ContractError::InvalidProofContextHash);
+        }
+
+        if (btp_header.main_height - signed_header.trusted_height) > state.trusting_period {
```

Comment:
We should also include a check:
assert(clientState.latestHeight - header.mainHeight < clientState.trustingPeriod)

Reply:
Yeah that should be fine. The generally should not have timeouts longer than that